os/migrating-to-clcs.md

```diff
@@ -0,0 +1,373 @@
+systemd:
+  units:
+    - name: "docker.service"
+      dropins:
+        - name: "50-insecure-registry.conf"
+          contents: |
+            [Service]
+            Environment=DOCKER_OPTS='--insecure-registry="10.0.1.0/24"'
+```
+
+Existing units can also be enabled without configuration.
+
+```container-linux-config
+systemd:
+  units:
+    - name: "etcd-member.service"
+      enable: true
+```
+
+### ssh_authorized_keys
+
+In a cloud config the `ssh_authorized_keys` section can be used to add ssh public keys to the `core` user.
+
+```cloud-config
+#cloud-config
+
+ssh_authorized_keys:
+  - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC0g+ZTxC7weoIJLUafOgrm+h..."
+```
+
+In a CLC there is no analogous section to `ssh_authorized_keys`, but ssh keys for the core user can be set just as easily using the `passwd.users.*` section:
+
+```container-linux-config
+passwd:
+  users:
+    - name: core
+      ssh_authorized_keys:
+        - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC0g+ZTxC7weoIJLUafOgrm+h..."
+```
+
+### hostname
+
+In a cloud config the `hostname` section can be used to set a machine's hostname.
+
+```cloud-config
+#cloud-config
+
+hostname: "coreos1"
+```
+
+There is no analogous section in a CLC, so to set the hostname a oneshot systemd unit can be added to call `hostnamectl`.
+
+```container-linux-config
+systemd:
+  units:
+    - name:   "set-hostname.service"
+      enable: true
+      contents: |
+        [Unit]
+        Description=set the machine's hostname
+        Type=oneshot
+
+        [Service]
+        ExecStart=hostnamectl set-hostname coreos1
+
+        [Install]
+        WantedBy=multi-user.target
+```
+
+### users
+
+The `users` section in a cloud config can be used to add users and specify many properties about them, from groups the user should be in to what the user's shell should be.
+
+```cloud-config
+#cloud-config
+
+users:
+  - name: "elroy"
+    passwd: "$6$5s2u6/jR$un0AvWnqilcgaNB3Mkxd5yYv6mTlWfOoCYHZmfi3LDKVltj.E8XNKEcwWm..."
+    groups:
+      - "sudo"
+      - "docker"
+    ssh-authorized-keys:
+      - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC0g+ZTxC7weoIJLUafOgrm+h..."
+```
+
+This same information can be added to the CLC in the `passwd.users.*` section.
+
+```container-linux-config
+passwd:
+  users:
+    - name:          "elroy"
+      password_hash: "$6$5s2u6/jR$un0AvWnqilcgaNB3Mkxd5yYv6mTlWfOoCYHZmfi3LDKVltj.E8XNKEcwWm..."
+      ssh_authorized_keys:
+        - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC0g+ZTxC7weoIJLUafOgrm+h..."
+      create:
+        groups:
+          - "sudo"
+          - "docker"
+```
+
+### write_files
+
+The `write_files` section in a cloud config can be used to specify files and their contents that should be written to disk on the machine.
+
+```cloud-config
+#cloud-config
+write_files:
+  - path:        "/etc/resolv.conf"
+    permissions: "0644"
+    owner:       "root"
+    content: |
+      nameserver 8.8.8.8
+```
+
+This can be done in a CLC with the `storage.files.*` section.
```

Comment:
Need to explain "filesystem" and we should also expand on contents (inline is used because blah).